### PR TITLE
added realtime update listener for user data

### PR DESCRIPTION
**Pull request by Kirazuto7**
Changes:

App.js: Includes event listen (onSnapshot) to capture updated user data in real time and saves them in states to pass to other components. Also added condition to prevent crash when user has not logged in and userId is unknown on app start.

Login.js & HomepageWithCards.js: Pseudo fixes navigation from login to homepage to display the rendered data after navigation by refreshing the page once after navigation.

### CODE REVIEW

File: my-app/src/Components/App.js
Comment (chizy7) [inline my-app/src/Components/App.js:1]: The changes to here seem reasonable and well-implemented. The addition of the event listener to capture updated user data in real-time is a great choice to improve the user experience. The use of states to pass user data to other components also appears well-implemented. Also great to see the code addresses a bug that could cause the app to crash when the user is not logged in and the user ID is unknown. I think this looks good.

File: my-app/src/Components/CardDeck.js
Comment (chizy7) [inline my-app/src/Components/CardDeck.js:1]: Looks good.

File: my-app/src/Components/Login.js
Comment (chizy7) [inline my-app/src/Components/Login.js:1]: Looks good

**Review by chizy7 — APPROVED** [review]
The changes looks good to me and makes sense what you are trying to do. Great job!

**Review by JChen0915 — APPROVED** [review]
Looks good, passing of props should make implementation of future features easier

**Pull request merged**